src/types/events/base-events.ts

```diff
@@ -34,27 +97,548 @@ export interface AppMentionEvent extends StringIndexed {
+      reply_count: number;
+      replies: MessageEvent[]; // TODO: should this be the union of all message events with type 'message'?
+      // TODO: unread_count doesn't appear in any other message event types, is this really the only place its included?
+      unread_count?: number;
+    };
+  };
+}
+
+export interface PinAddedEvent extends StringIndexed {
+  type: 'pin_added';
+  user: string;
+  channel_id: string;
+  // TODO: incomplete, what are all the types of items? probably message | file | file comment (deprecated)
+  item: {
+  };
+}
+
+export interface PinRemovedEvent extends StringIndexed {
+  type: 'pin_removed';
+  user: string;
+  channel_id: string;
+  // TODO: incomplete, what are all the types of items? probably message | file | file comment (deprecated)
+  item: {
+  };
+  has_pins: boolean;
+  event_ts: string;
+}
+
+export interface ReactionAddedEvent extends StringIndexed {
+  type: 'reaction_added';
+  user: string;
+  reaction: string;
+  item_user: string;
+  // TODO: incomplete, what are all the types of items? probably message | file | file comment (deprecated)
+  // https://api.slack.com/events/reaction_added
+  item: {
+  };
+  event_ts: string;
+}
+
+export interface ReactionRemovedEvent extends StringIndexed {
+  type: 'reaction_removed';
+  user: string;
+  reaction: string;
+  item_user: string;
+  // TODO: incomplete, what are all the types of items? probably message | file | file comment (deprecated)
+  // https://api.slack.com/events/reaction_removed
+  item: {
+  };
+  event_ts: string;
+}
+
+// NOTE: `resources_added`, `resources_removed`, `scope_denied`, `scope_granted`, are left out because they are
+// deprecated as part of the Workspace Apps Developer Preview
+
+export interface StarAddedEvent extends StringIndexed {
+  type: 'star_added';
+  user: string;
+  // TODO: incomplete, items are of type message | file | file comment (deprecated) | channel | im | group
+  // https://api.slack.com/events/star_added, https://api.slack.com/methods/stars.list
+  item: {
+  };
+  event_ts: string;
+}
+
+export interface StarRemovedEvent extends StringIndexed {
+  type: 'star_removed';
+  user: string;
+  // TODO: incomplete, items are of type message | file | file comment (deprecated) | channel | im | group
+  // https://api.slack.com/events/star_removed, https://api.slack.com/methods/stars.list
+  item: {
+  };
+  event_ts: string;
+}
+
+export interface SubteamCreated extends StringIndexed {
+  type: 'subteam_created';
+  // TODO: incomplete, this should probably be a reference to a Usergroup shape from @slack/types.
+  // https://api.slack.com/types/usergroup
+  subteam: {
+    id: string;
+  };
+}
+
+export interface SubteamMembersChanged extends StringIndexed {
+  type: 'subteam_members_changed';
+  subteam_id: string;
+  team_id: string;
+  date_previous_update: number;
+  date_update: number;
+  added_users: string[];
+  added_users_count: string; // are we sure this isn't a number?
```

Comment:
it's is, in fact, a number.

Reply:
cool, i'll file a ticket for this page then: https://api.slack.com/events/subteam_members_changed